DependencyInjection/Configuration.php

```diff
@@ -34,6 +34,14 @@ public function getConfigTreeBuilder()
                     ->prototype('scalar')
                     ->end()
                 ->end()
+                ->arrayNode('extractor')
+                    ->prototype('array')
```

Comment:
as you define a prototyped node for extractors, this still requires configuring extractors explicitly, otherwise you would get none. I would rather see a config like that:

```yaml
bazinga_js_translation:
    extraction:
        js: true # pre-configured extractor for JS, JSX and TS)
        coffee_script: false # separate service from JS with a different regex suited for the Coffee syntax as it may not have braces)
        custom_extractors:
            - { extensions: 'crazy', sequence: '->trans\(' }
```

This way, we can define a few built-in extractors that users just enable or disable, without having to care about what the regex is (and if they want a different config, they disable the built-in service and register a custom one)

Reply:
I'm kind of lost here.  How can I enable/disable dynamically a service from being instanciate by this config?
Or should I use this config to define the extractors in my own services configuration and delete the default one on this bundle?

Reply:
> delete the default one on this bundle

which default one ?

Reply:
The one with the id bazinga.jstranslation.translation_frontend_extractor from the Resources/config/services.xml

Reply:
all this configuration here is about defining some `bazinga.jstranslation.*` services (which is missing currently, as you don't process the config anymore).

The issue with your existing configuration tree is that you *only* have a config corresponding to my `custom_extractors` proposal, i.e. extractors configured in userland. So by default, there would be no extractor at all if they don't configure anything.

My proposed configuration would allow the bundle to manage itself the extensions and regex for the case of JS and CoffeeScript (and whatever pre-configured extractor we want to expose), while still allowing users to define more of them (I'm not even sure that second part is necessary, as they could just define some services themselves).

We must have some extractor service defined in the bundle. But we must have them working out of the box for users.

Reply:
I agree. More commits to come to fix this feature.